### DISCUSSION TITLE
Can buttons be changed to work On Click Release instead of On-Click?

### DISCUSSION THREAD

**Discussion by nekoalin** [TomSchimansky/CustomTkinter#1734] [Q&A]
Hello, 

Is there any way to change the CTKButtons to work on click release instead of on click, so that the user can move the mouse while clicked and release it somewhere else when pressed by mistake instead of having it function as soon as it's pressed?

Thank you!

**Comment by zSynctic** [comment]
hi, you should first make a CTkButton but don't include the command in its arguments instead bind the button with the buttonrelease which will call a function when the button is released. you should do something like this:

```
self.button = customtkinter.CTkButton(master=self)
self.button.bind("<ButtonRelease-1>", {functionname})
```

**Comment by avalon60** [comment]
> Hello,
> 
> Is there any way to change the CTKButtons to work on click release instead of on click, so that the user can move the mouse while clicked and release it somewhere else when pressed by mistake instead of having it function as soon as it's pressed?
> 
> Thank you!

Example:

```
import customtkinter as ctk

class App(ctk.CTk):
    def __init__(self):
        super().__init__()
        self.geometry("400x300")
        self.running = True

        button = ctk.CTkButton(self, text='Close')
        button.pack(padx=20, pady=20)
        button.bind("<ButtonRelease-1>", self.close_app)
        self.mainloop()

    def close_app(self, event):
        self.destroy()


if __name__ == "__main__":
    app = App()
```
Note that you need to specify an _event_ parameter in the callback function - `close_app(self, event)`.

**Reply by NazikBey** [reply]
Actually, this does not work, at least for me:

1. The button animation is not consistent with the click behaviour: shortly flashes instead of following click and possible draggging the mouse pointer out of the button location,
2. besides, the handler is triggered regardless of the mouse behaviour,
3. and finally, even if this had been a “solution”, the default behaviour should be fixed: a button action in always triggered on click release, not click down.

**Reply by FedericoSpada** [reply]
> Actually, this does not work, at least for me:
> 
> 1. The button animation is not consistent with the click behaviour: shortly flashes instead of following click and possible draggging the mouse pointer out of the button location,
> 2. besides, the handler is triggered regardless of the mouse behaviour,
> 3. and finally, even if this had been a “solution”, the default behaviour should be fixed: a button action in always triggered on click release, not click down.

Hi,
This change has been implemented in version 5.3.0, which is available with a different name on [PyPI](https://pypi.org/project/custom2kinter/), as reported in this [discussion](https://github.com/TomSchimansky/CustomTkinter/discussions/2343#discussioncomment-16532895).

You can try it right now by uninstalling `customtkinter` and installing `custom2kinter`. This new library is still imported using the original name, so existing code doesn't need to be updated.

**Comment by NazikBey** [comment]
> Le 23 avr. 2026 à 18:52, Federico Spada ***@***.***> a écrit :
> 
> Hi,
> This change has been implemented in version 5.3.0, which is available with a different name on PyPI <https://pypi.org/project/custom2kinter/>, as reported in this discussion <https://github.com/TomSchimansky/CustomTkinter/discussions/2343#discussioncomment-16532895>.
> 
> You can try it right now by uninstalling customtkinter and installing custom2kinter. This new library is still imported using the original name, so existing code doesn't need to be updated.
> 
> 

Thank you for your very quick answer.

Best,

NB
